sql/executor.go

```diff
@@ -39,11 +42,18 @@ var errTransactionInProgress = errors.New("there is already a transaction in pro
 type Executor struct {
 	db     client.DB
 	nodeID uint32
+
+	// System Config and mutex.
+	systemConfig   *config.SystemConfig
```

Comment:
One more color for the bikeshed: I picked up this pattern at GolangUK and like it for making the association between a mutex and its data more obvious:
```
type Executor struct {
    // other fields..

   systemConfig struct {
        sync.RWMutex
        value config.SystemConfig
   }
}
```

My preference is for mutex, atomic.Value, or unsafe.Pointer in that order.